zebra-test/src/mock_service.rs

```diff
@@ -0,0 +1,731 @@
+    ///
+    /// This determines the maximum amount of requests that are kept in queue before the oldest
+    /// request is dropped. This means that any tests that receive too many requests might ignore
+    /// some requests if this parameter isn't properly configured.
+    ///
+    /// The default value of 100 should be enough for most cases.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use zebra_test::mock_service::MockService;
+    /// #
+    /// let mock_service = MockService::build()
+    ///     .with_proxy_channel_size(100)
+    ///     .for_prop_tests();
+    /// #
+    /// # // Add types to satisfy the compiler's type inference.
+    /// # let typed_mock_service: MockService<(), (), _> = mock_service;
+    /// ```
+    pub fn with_proxy_channel_size(mut self, size: usize) -> Self {
+        self.proxy_channel_size = Some(size);
+        self
+    }
+
+    /// Configure the time to wait for a request before considering no requests will be received.
+    ///
+    /// This determines the maximum amount of time that the [`MockService`] will wait for a request
+    /// to be received before considering that a request will not be received.
+    ///
+    /// The default value of 25 ms should be enough for most cases.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use std::time::Duration;
+    /// #
+    /// # use zebra_test::mock_service::MockService;
+    /// #
+    /// let mock_service = MockService::build()
+    ///     .with_max_request_delay(Duration::from_millis(25))
+    ///     .for_unit_tests();
+    /// #
+    /// # // Add types to satisfy the compiler's type inference.
+    /// # let typed_mock_service: MockService<(), (), _> = mock_service;
+    /// ```
+    pub fn with_max_request_delay(mut self, max_request_delay: Duration) -> Self {
+        self.max_request_delay = Some(max_request_delay);
+        self
+    }
+
+    /// Create a [`MockService`] to be used in `proptest`s.
+    ///
+    /// The assertions performed by [`MockService`] use the macros provided by [`proptest`], like
+    /// [`prop_assert`].
+    pub fn for_prop_tests<Request, Response>(
+        self,
+    ) -> MockService<Request, Response, PropTestAssertion> {
+        self.finish()
+    }
+
+    /// Create a [`MockService`] to be used in Rust unit tests.
+    ///
+    /// The assertions performed by [`MockService`] use the macros provided by default in Rust,
+    /// like [`assert`].
+    pub fn for_unit_tests<Request, Response>(
+        self,
+    ) -> MockService<Request, Response, PanicAssertion> {
+        self.finish()
+    }
+
+    /// An internal helper method to create the actual [`MockService`].
+    ///
+    /// Note that this is used by both [`Self::for_prop_tests`] and [`Self::for_unit_tests`], the
+    /// only difference being the `Assertion` generic type parameter, which Rust infers
+    /// automatically.
+    fn finish<Request, Response, Assertion>(self) -> MockService<Request, Response, Assertion> {
```

Comment:
As we discussed we need to add `Error` here and in the above

Reply:
Done in [ad2c5c3](https://github.com/ZcashFoundation/zebra/pull/2748/commits/ad2c5c30a0e39408c613c50bb009f3f6e748d64b).